src/dmd/doc.d

```diff
@@ -393,7 +393,7 @@ extern(C++) void gendocfile(Module m)
             // BUG: convert file contents to UTF-8 before use
             const data = buffer.data;
             //printf("file: '%.*s'\n", cast(int)data.length, data.ptr);
-            mbuf.write(data.ptr, data.length);
+            mbuf.write(data[0 .. data.length]);
```

Comment:
`data` is already a slice

Reply:
ok

Reply:
I expect there are more of these, but this PR is about fixing the bottom. Fixing the next layer should be an additional PR.

I eventually want to make dmd @safe, and am starting with OutBuffer.